src/ray/raylet/agent_manager.h

```diff
@@ -70,7 +96,7 @@ class DefaultAgentManagerServiceHandler : public rpc::AgentManagerServiceHandler
   }
 
  private:
-  std::unique_ptr<AgentManager> &delegate_;
+  std::shared_ptr<AgentManager> &delegate_;
```

Comment:
why reference of shared_ptr, not just shared_ptr?

Reply:
I just use shared_ptr instead of unique_ptr. 
 It is because the construction of DefaultAgentManagerServiceHandler is happened before AgentManger construction.